frontend/pages/software/ManageSoftwarePage/ManageSoftwarePage.tsx

```diff
@@ -0,0 +1,395 @@
+      onSuccess: () => {
+        setIsLoadingSoftware(false);
+      },
+      onError: () => {
+        setIsLoadingSoftware(false);
+      },
+    }
+  );
+
+  const { data: softwareCount, error: softwareCountError } = useQuery<
+    ISoftwareCountResponse,
+    Error,
+    number
+  >(
+    ["softwareCount", { searchQuery, vulnerable: filterVuln, teamId }],
+    () => {
+      setIsLoadingCount(true);
+      return softwareAPI.count({
+        query: searchQuery,
+        vulnerable: filterVuln,
+        teamId,
+      });
+    },
+    {
+      keepPreviousData: true,
+      staleTime: 30000, // TODO: Discuss a reasonable staleTime given that counts are only updated infrequently?
+      refetchOnWindowFocus: false,
+      retry: 1,
+      select: (data) => data.count,
+      onSuccess: () => {
+        setIsLoadingCount(false);
+      },
+      onError: (err) => {
+        console.log("useQuery error: ", err);
+        setIsLoadingCount(false);
+      },
+    }
+  );
+
+  const onQueryChange = useDebouncedCallback(
+    async ({
+      pageIndex: newPageIndex,
+      searchQuery: newSearchQuery,
+      sortDirection: newSortDirection,
+      sortHeader: newSortHeader,
+    }: ITableQueryData) => {
+      pageIndex !== newPageIndex && setPageIndex(newPageIndex);
+      searchQuery !== newSearchQuery && setSearchQuery(newSearchQuery);
+      sortDirection !== newSortDirection &&
+        setSortDirection(
+          newSortDirection === "asc" || newSortDirection === "desc"
+            ? newSortDirection
+            : DEFAULT_SORT_DIRECTION
+        );
+      sortHeader !== newSortHeader && setSortHeader(newSortHeader);
+    },
+    300
+  );
+
+  const onTeamSelect = () => {
+    setPageIndex(0);
+  };
+
+  const renderHeaderButtons = (
+    state: ITeamsDropdownState
+  ): JSX.Element | null => {
+    if (
+      (state.isGlobalAdmin || state.isGlobalMaintainer) &&
+      state.teamId === 0
+    ) {
+      return (
+        <Button
+          onClick={() => console.log("Manage automations button click")}
+          className={`${baseClass}__manage-automations button`}
+          variant="brand"
+        >
+          <span>Manage automations</span>
+        </Button>
+      );
+    }
+    return null;
+  };
+
+  const renderHeaderDescription = (state: ITeamsDropdownState) => {
+    return (
+      <p>
+        Search for installed software and manage automations for detected
+        vulnerabilities (CVEs) on{" "}
+        <b>
+          {state.isPremiumTier && !!state.teamId
+            ? "all hosts assigned to this team"
+            : "all of your hosts"}
+        </b>
+        .
+      </p>
+    );
+  };
+
+  const renderHeader = useCallback(() => {
+    return (
+      <TeamsDropdownHeader
+        location={location}
+        router={router}
+        baseClass={baseClass}
+        onChange={onTeamSelect}
+        defaultTitle="Software"
+        description={renderHeaderDescription}
+        buttons={renderHeaderButtons}
+      />
```

Comment:
Is the idea to implement this on every page? If so, would you please create an issue tracking that and add to the Engineering board?

A nit, but when it is implemented on every page should it have a more general name than `TeamsDropdownHeader`? I'm thinking in a lot of cases it actually won't render a teams dropdown, so maybe something more universal like `PageHeader` makes sense?